compressed_image_transport/src/compressed_subscriber.cpp

```diff
@@ -47,41 +47,28 @@ using namespace cv;
 
 namespace enc = sensor_msgs::image_encodings;
 
+using CompressedImage = sensor_msgs::msg::CompressedImage;
+
 namespace compressed_image_transport
 {
 
-void CompressedSubscriber::subscribeImpl(ros::NodeHandle& nh, const std::string& base_topic, uint32_t queue_size,
-                             const Callback& callback, const ros::VoidPtr& tracked_object,
-                             const image_transport::TransportHints& transport_hints)
+void CompressedSubscriber::subscribeImpl(
+    rclcpp::Node::SharedPtr node,
+    const std::string& base_topic,
+    const Callback& callback,
+    rmw_qos_profile_t custom_qos)
 {
-    typedef image_transport::SimpleSubscriberPlugin<sensor_msgs::CompressedImage> Base;
-    Base::subscribeImpl(nh, base_topic, queue_size, callback, tracked_object, transport_hints);
-
-    // Set up reconfigure server for this topic
-    reconfigure_server_ = boost::make_shared<ReconfigureServer>(this->nh());
-    ReconfigureServer::CallbackType f = boost::bind(&CompressedSubscriber::configCb, this, _1, _2);
-    reconfigure_server_->setCallback(f);
+    typedef image_transport::SimpleSubscriberPlugin<CompressedImage> Base;
+    Base::subscribeImpl(node, base_topic, callback, custom_qos);
+    //TODO(ros2): Parameters or Dynamic reconfigure?
+    imdecode_flag_ = cv::IMREAD_UNCHANGED;
 }
 
 
-void CompressedSubscriber::configCb(Config& config, uint32_t level)
```

Comment:
not sure if it could help the future implementation of parameters to leave the configCb code as a comment

Reply:
I made a config struct that mirrors the config, I figure we can always jump back in the git log to get this.